src/main/java/org/wildfly/security/password/impl/BSDUnixDESCryptPasswordImpl.java

```diff
@@ -193,7 +193,7 @@ public int getSalt() {
         41, 52, 31, 37, 47, 55, 30, 40, 51, 45, 33, 48,
         44, 49, 39, 56, 34, 53, 46, 42, 50, 36, 29, 32
     };
-    private static final int[] invCompPerm = new int[56];
+    private static final byte[] invCompPerm = new byte[56];
```

Comment:
This seems to introduce potential sign-extension bugs, the first of which is at line 360 where a value from this array is used as an array index.  Where previously the value domain was (ostensibly) 0..255, as a byte[] the value domain becomes -128..127, which might result in a negative array index.

Because the result of most expressions in Java is an int, I think it's a good idea to use int for intermediate assignment types (as we are for inBit and outBit in that section).  But it does mean that every time we move a value from byte to int, we have to ensure that we're combining it with an ```& 0xff``` to avoid sign extension problems.